pkg/apis/policy/types.go

```diff
@@ -77,6 +77,10 @@ type PodDisruptionBudgetStatus struct {
 
 	// total number of pods counted by this disruption budget
 	ExpectedPods int32
+
+	// Conditions contain conditions for PDB
+	// +optional
+	Conditions []metav1.Condition
```

Comment:
this needs to be validated with a call to metav1validation.ValidateConditions

since this is a new field, it can be validated even for v1beta1 (which means actually connecting v1beta1 status updates to validation specifically for this field)

Reply:
I have added validation of the conditions to `ValidatePodDisruptionBudgetStatus`. This isn't actually used yet, but https://github.com/kubernetes/kubernetes/pull/99290 addresses this. I will make sure this validation gets run for all versions, while the existing fields will only be validated for v1.

Reply:
go ahead and wire the validation here... we don't want to merge unvalidated APIs assuming validation will be added in a follow-up before release

Reply:
Moved the code for wiring up status validation from https://github.com/kubernetes/kubernetes/pull/99290 into this PR.